docker/dev_env/entrypoint.sh

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env bash
+
+set -e
+
+if [ "$UID" = "0" ]; then
+  printf "Do not run this container with the root user! Use the 'run.sh' script or pass --user to docker run.\n"
+  exit 1
+fi
+
+if [ ! -d "$OPENVERSE_PROJECT"/.git ]; then
+  printf "Repository not mounted to container!\n"
+  exit 1
+fi
+
+cd "$OPENVERSE_PROJECT" || exit 1
+
+suppress_output() {
+  "$@" 2>/dev/null 1>/dev/null
+}
+
+suppress_output corepack install
+
+if [ -z "$(n ls 2>/dev/null)" ]; then
+  printf "Installing the specific Node JS version required by Openverse frontend; this is only necessary the first time the toolkit runs\n"
+  n install auto
+fi
+
+if [ -n "$PNPM_HOME" ]; then
+  export PATH="$PNPM_HOME:$PATH"
+fi
+
+_python3s=("$HOME"/.local/share/pdm/python/cpython@3.11.*/bin/python3)
+
+if [ ! -x "${_python3s[0]}" ]; then
+  printf "Installing the specific Python version required for pipenv environments; this is only necessary the first time the toolkit runs\n"
+  pdm python install 3.11
+  _python3s=("$HOME"/.local/share/pdm/python/cpython@3.11.*/bin/python3)
+fi
+
+PYTHON_311=${_python3s[0]}
+export PYTHON_311
+
+if [ ! -x "$HOME"/.local/bin/python3.11 ]; then
+  ln -s "$PYTHON_311" "$HOME"/.local/bin/python3.11
+fi
+
+if [ -z "$(command -v pipenv)" ]; then
+  # Install pipenv with the specific python version required by environments
+  # still using it, otherwise it gets confused about which dependencies to use
+  pipx install pipenv --python "$PYTHON_311"
+fi
+
+if [ -n "$PDM_CACHE_DIR" ]; then
+  pdm config install.cache on
+fi
+
+_profiles=(
+  "$OPENVERSE_PROJECT"/.ovprofile
+  "$OPENVERSE_PROJECT"/docker/dev_env/bash_profile.sh
+)
+
+for profile in "${_profiles[@]}"; do
+  if [ -f "$profile" ]; then
+    if grep -q '^alias ' "$profile"; then
+      printf "Aliases in %s will not work. Use functions instead." "$profile" >>/dev/stderr
+      exit 1
+    fi
+
+    # SC1090 cannot be disabled directly
+    # Workaround from https://github.com/koalaman/shellcheck/issues/2038#issuecomment-680513830
+    # shellcheck source=/dev/null
+    source "$profile"
+  fi
+done
+
+"$@"
```

Comment:
I'm basing this on the API `run.sh` entrypoint-file but I think `exec` is needed here?
```suggestion
exec "$@"
```

Reply:
It isn't strictly necessary, the array syntax (`$@`) expands correctly in bash for this to work. It would be necessary if it was `$*` instead.

Fine to change though, doesn't hurt anything.

Reply:
I have read that `exec` is needed for `SIG*` to be passed correctly because it replaces the process with the contents of `"$@"`. It's some nuance that I don't fully understand but let's change it just for consistency, if nothing else.

Reply:
Ahhh, good to know that! I'll have to consider that for other places.

I've already applied your suggestion in the latest, so we'll be good to go :slightly_smiling_face: